### ISSUE TITLE
C++ specialization: warning: explicit link request could not be resolved.

### ISSUE THREAD

**Issue by q-p** [doxygen/doxygen#10964]
**Describe the bug**
In the following C++ code, the method linked to in the specialization cannot be found.
```cpp
#include <type_traits>

/// Generic template
template<class T, class Enabled = void>
class Foo { };

/// Specialization.
/**
 * Link to #Bar(). This one is not found.
 */
template<class T>
class Foo<
  T, 
  std::enable_if_t<std::is_base_of<T, typename T::Type>::value>>
{
public:
  /// Computes stuff, another #Bar() link, also not found and reported twice.
  void Bar() const { }
};
```

**To Reproduce**
Reproducer
[warning_explicit_link.zip](https://github.com/user-attachments/files/15984769/warning_explicit_link.zip)

**Expected behavior**
I think the link to `Bar()` should be found and resolved.

**Version**
The following reproduces the warning on both 1.11 and current master (6221f43137ebb2c3daa5cffae807105825612b8b). Both installed using homebrew on Mac.

**Additional context**
It seems to work if in the `enable_if_t` I replace `std::is_base_of<T, typename T::Type>::value` by `std::is_base_of_v<T, typename T::Type>`, but unfortunately in the "real" code, no `_v` version of the problematic trait exists.

**Comment by albert-github** [comment]
The fact that doxygen reports some warnings multiple times is due to the fact that some comment blocks are used multiple times and are processed multiple times. Doxygen does not keep a stack of used warnings, hence the double warnings.

**Comment by doxygen** [comment]
@q-p Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by q-p** [comment]
I tried master as of 4a9be1303 and that does indeed fix the problem: The warnings are gone and the links resolve as expected. Thanks!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
